src/nn/activations.rs

```diff
@@ -55,9 +60,175 @@ where
     }
 }
 
+/// Calls [prelu()] with constant value.
+#[derive(Debug, Clone, Copy)]
+pub struct LeakyReLU<E: Dtype>(E);
+
+impl<E: Dtype> Default for LeakyReLU<E> {
+    fn default() -> Self {
+        Self(E::from_f32(0.05).unwrap())
+    }
+}
+
+impl<E: Dtype> ZeroSizedModule for LeakyReLU<E> {}
+impl<E: Dtype> NonMutableModule for LeakyReLU<E> {}
+
+impl<S: ConstShape, E: Dtype, D: Device<E>, T: Tape<E, D>> Module<Tensor<S, E, D, T>>
+    for LeakyReLU<E>
+where
+    Tensor<S, E, D, T>: TryPReLU<E, Output = Tensor<S, E, D, T>>,
+{
+    type Output = Tensor<S, E, D, T>;
+    type Error = D::Err;
+
+    fn try_forward(&self, input: Tensor<S, E, D, T>) -> Result<Self::Output, D::Err> {
+        Ok(leakyrelu(input, self.0))
+    }
+}
+
+pub mod builder {
+    #[derive(Debug, Copy, Clone, Eq, PartialEq)]
+    pub struct PReLU;
+}
+
+impl<E: Dtype, D: Device<E>> BuildOnDevice<D, E> for builder::PReLU
+where
+    PReLU<E, D>: BuildModule<D, E>,
+{
+    type Built = PReLU<E, D>;
+    fn try_build_on_device(device: &D) -> Result<Self::Built, <D>::Err> {
+        Self::Built::try_build(device)
+    }
+}
+
+/// Calls [prelu()] with learnable value.
+#[derive(Debug, Clone)]
+pub struct PReLU<E: Dtype, D: Device<E>> {
+    a: Tensor<(), E, D>,
+}
+
+impl<E: Dtype, D: Device<E>> Default for PReLU<E, D> {
+    fn default() -> Self {
+        let dev = D::default();
+        Self {
+            a: dev.tensor(E::from_f32(0.25).unwrap()),
+        }
+    }
+}
+
+impl<E: Dtype, D: Device<E>> From<E> for PReLU<E, D> {
+    fn from(value: E) -> Self {
+        let dev = D::default();
+        Self {
+            a: dev.tensor(value),
+        }
+    }
+}
```

Comment:
Let's remove these so it's consistent with other nn modules - so it has to go through BuildModule to get access to device

Reply:
done